src/GraphQL/Types/Scalars/ScalarGraphType.cs

```diff
@@ -20,9 +20,10 @@ public abstract class ScalarGraphType : GraphType
         /// responsible for preparing the scalar for transport to the client. It is only responsible
         /// for generating an object which can eventually be serialized by some transport-focused API.
         /// <br/><br/>
-        /// This method must handle a value of <see langword="null"/>.
+        /// This method should handle a value of <see langword="null"/>, but may throw an exception
+        /// if <see langword="null"/> is an invalid internal scalar representation.
         /// </summary>
-        /// <param name="value">Resolved value (internal scalar representation). May be <see langword="null"/>.</param>
+        /// <param name="value">Resolved value (internal scalar representation). Might be <see langword="null"/> if returned from a field resolver.</param>
```

Comment:
?

Reply:
I realized that it is the AST representation that is enforced by the graphql spec.  The spec does not control what internal values might be.  So when writing the sample above where null external values are internal (db) 0 values, I realized it was entirely valid for Serialize to throw an error upon `Serialize(null)`.

I also considered the reverse logic -- can a scalar refuse an AST with a value of NullValue?  Well maybe, actually, if that is never a valid value for the scalar.  But AST nodes must support null, as they are coerced to null when errors occur and so on.  And if they should not support null, it should be represented as a non-null graph type -- not a scalar that throws an exception when it's passed NullValue.  So I only changed Serialize.  Thoughts?

Reply:
I had more questions when I saw this change. `value` maybe `null`? Yes. This is a completely understandable description without any additional `if...`.